src/server/frontend_wayland/text_input_v3.cpp

```diff
@@ -331,11 +332,26 @@ void mf::TextInputV3::send_text_change(ms::TextInputChange const& change)
                 auto scan_code = change.keysym->sym;
                 switch (scan_code)
                 {
+                    case XKB_KEY_BackSpace:
+                        scan_code = KEY_BACKSPACE;
+                        break;
                     case XKB_KEY_Return:
-                        // TODO: kwin_wayland maps this to 28, but I am unsure how they get there
-                        scan_code = 28;
+                        scan_code = KEY_ENTER;
+                        break;
+                    case XKB_KEY_Left:
+                        scan_code = KEY_LEFT;
+                        break;
+                    case XKB_KEY_Right:
+                        scan_code = KEY_RIGHT;
+                        break;
+                    case XKB_KEY_Up:
+                        scan_code = KEY_UP;
+                        break;
+                    case XKB_KEY_Down:
+                        scan_code = KEY_DOWN;
                         break;
                     default:
+                        scan_code = KEY_UNKNOWN;
```

Comment:
That default is different from previous versions. Intentional?

Reply:
Yes it appears to be the case. `kwin` thinks that the defaults in the cases are the _only_ keysym events that should be sent via input method